src/main/java/net/spy/memcached/protocol/ascii/TouchOperationImpl.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * arcus-java-client : Arcus Java client
+ * Copyright 2010-2014 NAVER Corp.
+ * Copyright 2014-present JaM2in Co., Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.spy.memcached.protocol.ascii;
+
+import net.spy.memcached.KeyUtil;
+import net.spy.memcached.ops.OperationCallback;
+import net.spy.memcached.ops.OperationState;
+import net.spy.memcached.ops.OperationStatus;
+import net.spy.memcached.ops.StatusCode;
+import net.spy.memcached.ops.TouchOperation;
+
+import java.nio.ByteBuffer;
+import java.util.Collection;
+import java.util.Collections;
+
+/**
+ * Memcached touch operation.
+ */
+final class TouchOperationImpl extends OperationImpl implements TouchOperation {
+
+  private static final int OVERHEAD = 9;
+
+  private static final OperationStatus OK = new OperationStatus(true, "TOUCHED",
+          StatusCode.SUCCESS);
+  private static final OperationStatus NOT_FOUND = new OperationStatus(false, "NOT_FOUND",
+          StatusCode.ERR_NOT_FOUND);
+
+  private final String key;
+  private final int exp;
+
+  public TouchOperationImpl(String k, int t, OperationCallback cb) {
+    super(cb);
+    key = k;
+    exp = t;
+  }
+
+  public Collection<String> getKeys() {
+    return Collections.singleton(key);
+  }
+
+  @Override
+  public void handleLine(String line) {
+    if(line.equals("TOUCHED")) {
+      getLogger().debug("Touch completed successfully");
+      getCallback().receivedStatus(matchStatus(line, OK));
+    } else if(line.equals("NOT_FOUND")) {
+      getLogger().debug("Touch key is Not found");
+      getCallback().receivedStatus(matchStatus(line, NOT_FOUND));
+    }
+    transitionState(OperationState.COMPLETE);
```

Comment:
@uhm0311 @jhpark816 
replication, migration 고려하면 아래와 같은 형태가 되어야 할 것 같은데 확인 부탁드립니다.

```suggestion
    /* ENABLE_REPLICATION if */
    if (hasSwitchedOver(line)) {
      prepareSwitchover(line);
      return;
    }
    /* ENABLE_REPLICATION end */
    /* ENABLE_MIGRATION if */
    if (hasNotMyKey(line)) {
      addRedirectSingleKeyOperation(line, key);
      transitionState(OperationState.REDIRECT);
      return;
    }
    /* ENABLE_MIGRATION end */
    
    complete(matchStatus(line, OK, NOT_FOUND));
```